commands/addons/attach.js

```diff
@@ -23,12 +25,8 @@ function * run (context, heroku) {
     )
   }
 
-  let attachment = yield createAttachment(app, context.flags.as, context.flags.confirm)
-    .catch((err) => {
-      if (!err.body || err.body.id !== 'confirmation_required') throw err
-      return cli.confirmApp(app, context.flags.confirm, err.body.message)
-        .then(() => createAttachment(app, context.flags.as, app))
-    })
+  let attachment = yield util.trapConfirmationRequired(context, (confirm) => createAttachment(app, context.flags.as, confirm))
```

Comment:
@dickeyxxx I am open to refactoring this if it would make it clearer, though I am not sure how your example would work, as the confirmationRequired needs a way to run the query again after the confirm block has finished successfully.